app/api.py

```diff
@@ -297,6 +301,37 @@ def get(self):
             self, ApiActionLog.READ, len(records), len(notes))
 
 
+class PhotoUpload(utils.BaseHandler):
+    https_required = True
+
+    def post(self):
+        if not (self.auth):
+            self.info(
+                403,
+                message='Missing or invalid authorization key',
+                style='plain')
+            return
+        photo = self.request.body_file
+        if photo is not None:
+            photo = photo.read()
```

Comment:
Using the same variable "photo" for various types (file, string, Image) is confusing. Can you use different variable names for each of them?